src/basic/fileio.c

```diff
@@ -645,9 +647,12 @@ int read_full_stream_full(
                 n_next = size != SIZE_MAX && !FLAGS_SET(flags, READ_FULL_FILE_FAIL_WHEN_LARGER) ? size : LINE_MAX;
 
         /* Never read more than we need to determine that our own limit is hit */
-        if (n_next > READ_FULL_BYTES_MAX)
+        if (n_next > READ_FULL_BYTES_MAX && !FLAGS_SET(flags, READ_FULL_FILE_LARGE_FILE))
                 n_next = READ_FULL_BYTES_MAX + 1;
 
+        if (n_next > READ_FULL_LARGE_BYTES_MAX)
+                n_next = READ_FULL_LARGE_BYTES_MAX + 1;
```

Comment:
That is ugly. Maybe:
```C

const size_t limit = FLAGS_SET(flags, READ_FULL_FILE_LARGE_FILE) ? READ_FULL_LARGE_BYTES_MAX :  READ_FULL_BYTES_MAX);

n_next = MIN(n_next, limit + 1);
```

and then limit can also be reused later on, instead of repeating the same condition.

Reply:
yes, done